### PR TITLE
Fish completions

### PR DESCRIPTION
**Pull request by rpowelll**
This pull request adds auto completion of Carthage commands and options for the [fish shell](http://fishshell.com).

### CODE REVIEW

File: Documentation/BashZshFishCompletion.md [hunk 1]
```diff
@@ -41,9 +42,21 @@ compinit -u
...
 
 ```
-# Check $fpath
+# check $fpath
 echo $fpath
 
-# Create a symbolic link
-ln -s /Library/Frameworks/CarthageKit.framework/Versions/A/Scripts/carthage-zsh-completion /path/to/fpath/directory/_carthage
+# create a symbolic link
+ln -s /library/frameworks/carthagekit.framework/versions/a/scripts/carthage-zsh-completion /path/to/fpath/directory/_carthage
```
Comment (nanoxd) [inline Documentation/BashZshFishCompletion.md]: Path needs to be capitalized to work properly :smile:

File: Documentation/BashZshFishCompletion.md [hunk 2]
```diff
@@ -41,9 +42,21 @@ compinit -u
...
+### Fish
+
+Create a symbolic link named `carthage.fish` in one of the directories specified by `$fish_complete_path`, and point it to `/Library/Frameworks/CarthageKit.framework/Versions/A/Scripts/carthage-fish-completion`.
+
+```
+# check $fish_complete_path
+echo $fish_complete_path
+
+# create a symbolic link
+ln -s /library/frameworks/carthagekit.framework/versions/a/scripts/carthage-fish-completion /path/to/fpath/directory/carthage.fish
```
Comment (nanoxd) [inline Documentation/BashZshFishCompletion.md]: Path also needs to be capitalized here :8ball:

File: Source/Scripts/carthage-fish-completion [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  return 1
+end
+
+complete -c carthage -n '__fish_prog_needs_subcommand' -a "archive bootstrap build checkout copy-frameworks fetch help update version" -f
+
+complete -c carthage -n '__fish_prog_using_subcommand archive' -l output -r
+complete -c carthage -n '__fish_prog_using_subcommand archive' -l color -a "auto always never" -x
+
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l configuration -r
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l platform -x -a "all Mac iOS watchOS tvOS"
```
Comment (nanoxd) [inline Source/Scripts/carthage-fish-completion]: `OSX` should be added to the list per #910

File: Source/Scripts/carthage-fish-completion [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l verbose
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l no-checkout
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l no-build
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l use-ssh
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l use-submodules
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l no-use-binaries
+complete -c carthage -n '__fish_prog_using_subcommand bootstrap' -l color -a "auto always never" -x
+
+complete -c carthage -n '__fish_prog_using_subcommand build' -l configuration -r
+complete -c carthage -n '__fish_prog_using_subcommand build' -l platform -x -a "all Mac iOS watchOS tvOS"
```
Comment (nanoxd) [inline Source/Scripts/carthage-fish-completion]: Same as above

File: Source/Scripts/carthage-fish-completion [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+complete -c carthage -n '__fish_prog_using_subcommand checkout' -l use-submodules
+complete -c carthage -n '__fish_prog_using_subcommand checkout' -l no-use-binaries
+complete -c carthage -n '__fish_prog_using_subcommand checkout' -l color -a "auto always never" -x
+
+complete -c carthage -n '__fish_prog_using_subcommand fetch' -l color -a "auto always never" -x
+
+complete -c carthage -n '__fish_prog_using_subcommand help' -a "archive bootstrap build checkout copy-frameworks fetch help update version" -f
+
+complete -c carthage -n '__fish_prog_using_subcommand update' -l configuration -r
+complete -c carthage -n '__fish_prog_using_subcommand update' -l platform -x -a "all Mac iOS watchOS tvOS"
```
Comment (nanoxd) [inline Source/Scripts/carthage-fish-completion]: Same as above

File: Documentation/BashZshFishCompletion.md (hunk 1 above)
Comment (rpowelll) [inline Documentation/BashZshFishCompletion.md]: Whoops, not sure how that happened, probably bumped the ~ key in vim

Comment (nanoxd) [inline Documentation/BashZshFishCompletion.md]: :+1:

**Comment by rpowelll** [comment]
Alright, all those issues have been fixed up.

**Comment by nanoxd** [comment]
@rpowelll looks like a049053 did not cover adding `OSX` to the `build` command completion

**Comment by rpowelll** [comment]
Fixed!

**Comment by mdiep** [comment]
Thank you for contributing this!

**Pull request merged**